### PR TITLE
[snapshot] excludes helpers in assets when finding helpers

### PR DESCRIPTION
**Pull request by YusukeHosonuma**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
When install fastlane with `bundle install --path xxx` (e.g. `bundle install --path vendor/bundle`), snapshot helpers is created under the current directory. In current `fastlane snapshot update` is check the all sources under the current directory. Therefore, those sources are also included. 

```
$ bundle install --path vendor/bundle
$ bundle exec fastlane snapshot update
[18:29:12]: Found the following SnapshotHelper:
[18:29:12]:     ./sample-ios-swift-snapshot/Helpers/SnapshotHelper.swift
[18:29:12]:     ./vendor/bundle/ruby/2.4.0/gems/fastlane-2.54.3/snapshot/lib/assets/SnapshotHelper.swift
[18:29:12]:     ./vendor/bundle/ruby/2.4.0/gems/fastlane-2.54.3/snapshot/lib/assets/SnapshotHelperXcode8.swift
[18:29:12]: Are you sure you want to automatically update the helpers listed above?
[18:29:12]: This will overwrite all its content with the latest code.
[18:29:12]: The underlying API will not change. You can always migrate manually by looking at
[18:29:12]: https://github.com/fastlane/fastlane/blob/master/snapshot/lib/assets/SnapshotHelper.swift
[18:29:12]: Overwrite configuration files? (y/n)
```

Helpers under `lib/assets` is original sources, therefore it should not be included in lists. Expected behaviors are as follows.

```
$ bundle exec fastlane snapshot update
[18:30:33]: Found the following SnapshotHelper:
[18:30:33]:     ./sample-ios-swift-snapshot/Helpers/SnapshotHelper.swift
[18:30:33]: Are you sure you want to automatically update the helpers listed above?
[18:30:33]: This will overwrite all its content with the latest code.
[18:30:33]: The underlying API will not change. You can always migrate manually by looking at
[18:30:33]: https://github.com/fastlane/fastlane/blob/master/snapshot/lib/assets/SnapshotHelper.swift
[18:30:33]: Overwrite configuration files? (y/n)
```

### Description
Excludes helper paths in assets when finding helpers.

### CODE REVIEW

**Comment by ohayon** [comment]
Hey @YusukeHosonuma, thanks for the PR! I'm curious, what is the implication of the current implementation? Shouldn't it still find the Helper file? 🐙

**Comment by YusukeHosonuma** [comment]
@ohayon Thanks for response!

I'm sorry that the explanation is missing.

In current, Helpers are two versions exists (For Xcode 9  and Xcode 8).

[SnapshotHelper.swift (for Xcode 9)](https://github.com/fastlane/fastlane/blob/master/snapshot/lib/assets/SnapshotHelper.swift#L193-195)
```swift
...
// Please don't remove the lines below
// They are used to detect outdated configuration files
// SnapshotHelperVersion [1.5]
```

[SnapshotHelperXcode8.swift](https://github.com/fastlane/fastlane/blob/master/snapshot/lib/assets/SnapshotHelperXcode8.swift#L171-173)
```swift
...
// Please don't remove the lines below
// They are used to detect outdated configuration files
// SnapshotHelperXcode8Version [1.4]
```

When execute `fastlane snapshot`, validate Helper version.
It validates all Helpers under current directory.

For example if use Xcode 8 project, valid version is `SnapshotHelperXcode8Version [1.4]`.
When install fastlane by `bundle install --path vendor/bundle` then execute `bundle exec fastlane snapshot`, get unexpected result.

```
$ bundle exec fastlane snapshot
[21:23:11]: Successfully loaded '/Users/yusuke/Desktop/FastlaneSnapshotExample/fastlane/Snapfile'
...
[21:23:13]: Your './vendor/bundle/ruby/2.4.0/gems/fastlane-2.55.0/snapshot/lib/assets/SnapshotHelper.swift' is outdated, please run `fastlane snapshot update`
[21:23:13]: to update your Helper file

[!] Please update your Snapshot Helper file
```

The same problem also occurs in use Xcode 9.

```
[00:25:52]: Your './vendor/bundle/ruby/2.4.0/gems/fastlane-2.55.0/snapshot/lib/assets/SnapshotHelperXcode8.swift' is outdated, please run `fastlane snapshot update`
[00:25:52]: to update your Helper file

[!] Please update your Snapshot Helper file
```

In other words, if installing fastlane using `bundle install --path vendor/bundle`, `snapshot` fails.
Because validate Helpers under `vendor/bundle/**/lib/assets/`.

That's why should excludes Helpers under `lib/assets`.

I'm sorry, my English is not good.
Thank you.

**Comment by YusukeHosonuma** [comment]
Hi @ohayon 

I've commented that why should exclude Helpers sources under `lib/assets/` at execute `fastlane snapshot`.
Please tell me if it’s difficult to understand.

Thanks.

**Review by ohayon — APPROVED** [review]
👍 Thanks @YusukeHosonuma !!

**Comment by ohayon** [comment]
Just noticed this is your first contribution to _fastlane_, welcome to the community!! 🚀 🐙
